src/include/txn_inline.h

```diff
@@ -692,21 +708,57 @@ __wt_txn_visible_all(WT_SESSION_IMPL *session, uint64_t id, wt_timestamp_t times
     return (pinned_ts != WT_TS_NONE && timestamp <= pinned_ts);
 }
 
+/*
+ * __wt_txn_upd_get_durable --
+ *     Safely retrieve a durable timestamp from an update.
+ */
+static inline int
+__wt_txn_upd_get_durable(WT_SESSION_IMPL *session, WT_UPDATE *upd, wt_timestamp_t *durable_ts)
+{
+    uint8_t prepare_state;
+
+    WT_ORDERED_READ(prepare_state, upd->prepare_state);
+
+    WT_ASSERT_ALWAYS(session,
```

Comment:
If we can indeed assert here it'll be a lot easier to change this return to be the `durable_ts` itself. Additionally I would expect the setter function to be `void`.

How did you confirm that an assertion is valid here?

Reply:
@ajmorton if we can assert on prepared state here. Does it make sense to order the read? I haven't had enough time to think about this but needs considering.